code/processes/reporter.q

```diff
@@ -116,8 +116,8 @@ runreport:{[tab]
 		`..timerids upsert 1!select id,periodend:nextrun from .timer.timer where fp~/:funcparam;
 	:()];
         / - if timezone is specified convert the start and end timestamps to the time used by the system 
-    startts:?[not null tab[`timezone];.tz.ttz[`GMT;tab[`timezone];gtime[startts]];startts];
-    endts:?[not null tab[`timezone];.tz.ttz[`GMT;tab[`timezone];gtime[endts]];endts];
+    startts:?[not null tab[`timezone];.tz.ttz[`GMT;tab[`timezone];?[`localtime in .proc.params;gtime[startts];startts]];startts];
+    endts:?[not null tab[`timezone];.tz.ttz[`GMT;tab[`timezone];?[`localtime in .proc.params;gtime[endts];endts]];endts];
```

Comment:
I think you can just check `.proc.localtime` here, and probably use `$` instead of `?`

```suggestion
    startts:?[not null tab[`timezone];.tz.ttz[`GMT;tab[`timezone];$[.proc.localtime;gtime[startts];startts]];startts];
    endts:?[not null tab[`timezone];.tz.ttz[`GMT;tab[`timezone];$[.proc.localtime;gtime[endts];endts]];endts];
```